docgen/src/guide/React_native.md

```diff
@@ -6,9 +6,643 @@ category: guide
+            }}
+            >
+            <SearchBox />
+            </View>
+            <Hits />
+        </InstantSearch>
+    </View>
+```
+
+
+The search is now interactive but we don't see what matched in each of the products.
+
+Good thing for us, Algolia computes the matching part. 
+
+For taking  advantage of this feature, we need to use the [connectHighlight](connectors/connectHighlight.html) connector. Let’s create a new `<Highlight>` component. 
+
+This connector gives you one interesting properties: 
+
+* `highlight`, a function that returns an array of parsed hits indicating if they are highlighted or not. 
+
+When using this component you will have to provide two properties: 
+
+* `attributeName` attribute name from the record that you want to highlight. 
+* `hit` , the hit to highlight
+
+```jsx 
+//We need to the connectHighlight to our import
+import {
+  connectInfiniteHits,
+  connectSearchBox,
+  connectHighlight,
+} from 'react-instantsearch/connectors';
+
+const Highlight = connectHighlight(
+  ({ highlight, attributeName, hit, highlightProperty }) => {
+    const parsedHit = highlight({
+      attributeName,
+      hit,
+      highlightProperty: '_highlightResult',
+    });
+    const highligtedHit = parsedHit.map((part, idx) => {
+      if (part.isHighlighted)
+        return (
+          <Text key={idx} style={{ backgroundColor: '#ffff99' }}>
+            {part.value}
+          </Text>
+        );
+      return part.value;
+    });
+    return <Text>{highligtedHit}</Text>;
+  }
+);
+```
+
+Let’s use this new `<Highlight/>` component inside our `<Hits/>` template: 
+
+```jsx
+// [...]
+
+const Hits = connectInfiniteHits(({ hits, hasMore, refine }) => {
+  
+ // [...]
+
+  return (
+    <FlatList
+      data={hits}
+      onEndReached={onEndReached}
+      keyExtractor={keyExtractor}
+      renderItem={({ item }) => {
+        return (
+          <View style={{ flexDirection: 'row', alignItems: 'center' }}>
+            <Image
+              style={{ height: 100, width: 100 }}
+              source={{ uri: item.image }}
+            />
+            <View style={{ flex: 1 }}>
+              <Text>
+                <Highlight attributeName="name" hit={item} />
+
+              </Text>
+              <Text>
+                <Highlight attributeName="type" hit={item} />
+
+              </Text>
+            </View>
+          </View>
+        );
+      }}
+    />
+  );
+});
+```
+
+Now the search displays the results and highlights the part of the hit attribute that matches the query. This pattern is very important in search, especially with Algolia, so that the user knows what's going on. This way we create the search dialog between the user and the data.
+
+In this part, we've seen the following:
+- how to add a SearchBox to make queries into the records
+- how to highlight the matched part of the results
+- the importance of highlighting in a text-based search
+
+## Add RefinementList
```

Comment:
Also it implies that we may have a RefinementList widget for native, while we don't